server/services/withings/lib/withings.poll.js

```diff
@@ -0,0 +1,218 @@
+const Promise = require('bluebird');
+const uuid = require('uuid');
+const { DEVICE_FEATURE_TYPES } = require('../../../utils/constants');
+const OAuth2Manager = require('../../../lib/oauth2');
+const logger = require('../../../utils/logger');
+
+/**
+ * @description Poll value of a withings device.
+ * @param {Object} device - The device to update value.
+ * @returns {Promise} Resolve with withings device poll.
+ * @example
+ * poll(device);
+ */
+async function poll(device) {
+  logger.debug('Device to poll:', device);
+  if (device.features) {
+    const oauth2Manager = new OAuth2Manager(this.gladys);
+    const withingsDeviceId = device.params.find((oneParam) => oneParam.name === 'withingsDeviceId').value;
+
+    logger.debug('Features : ', device.features);
+
+    const { gladys } = this;
+    const { serviceId } = this;
+
+    // Get all users of gladys
+    const users = await gladys.user.get();
+    await Promise.map(
+      users,
+      async (user) => {
+        const withingsClienId = await gladys.variable.getValue('WITHINGS_CLIENT_ID', serviceId, user.id);
+
+        if (withingsClienId) {
+          device.features.forEach(async (feat) => {
```

Comment:
Idem foreach with async/await...

Reply:
I remove async declaration

Reply:
FYI, I found these articles where they explain why "forEach is not promise-aware": 
https://zellwk.com/blog/async-await-in-loops/
https://advancedweb.hu/how-to-use-async-functions-with-array-foreach-in-javascript/